usr/lib/cca_stdll/cca_specific.c

```diff
@@ -4131,21 +4359,83 @@ static CK_RV ccatok_wrap_key_rsa_pkcs(CK_MECHANISM *mech, CK_BBOOL length_only,
     case CKK_DES:
     case CKK_DES2:
     case CKK_DES3:
-        rule_array_count = 2;
         switch (mech->mechanism) {
         case CKM_RSA_PKCS:
+            rule_array_count = 2;
             memcpy(rule_array, "DES     PKCS-1.2", 2 * CCA_KEYWORD_SIZE);
             break;
+        case CKM_RSA_PKCS_OAEP:
+            rule_array_count = 3;
+            oaep = (CK_RSA_PKCS_OAEP_PARAMS *)mech->pParameter;
+            if (oaep == NULL ||
+                mech->ulParameterLen != sizeof(CK_RSA_PKCS_OAEP_PARAMS))
+                return CKR_MECHANISM_PARAM_INVALID;
+
+            if (oaep->source == CKZ_DATA_SPECIFIED &&
+                oaep->ulSourceDataLen > 0) {
+                TRACE_ERROR("CCA does not support non-empty OAEP source "
+                            "data\n");
+                return CKR_MECHANISM_PARAM_INVALID;
+            }
+
+            switch (oaep->hashAlg) {
+            case CKM_SHA_1:
+                if (oaep->mgf != CKG_MGF1_SHA1)
+                    return CKR_MECHANISM_PARAM_INVALID;
+                memcpy(rule_array, "DES     PKCSOAEPSHA-1   ",
+                       3 * CCA_KEYWORD_SIZE);
+                break;
+            case CKM_SHA256:
+                if (oaep->mgf != CKG_MGF1_SHA256)
+                    return CKR_MECHANISM_PARAM_INVALID;
+                memcpy(rule_array, "DES     PKCSOAEPSHA-256 ",
+                       3 * CCA_KEYWORD_SIZE);
+                break;
+            default:
+                return CKR_MECHANISM_PARAM_INVALID;
+            }
+            break;
         default:
             return CKR_MECHANISM_INVALID;
         }
         break;
     case CKK_AES:
-        rule_array_count = 2;
         switch (mech->mechanism) {
         case CKM_RSA_PKCS:
+            rule_array_count = 2;
             memcpy(rule_array, "AES     PKCS-1.2", 2 * CCA_KEYWORD_SIZE);
             break;
+        case CKM_RSA_PKCS_OAEP:
+            rule_array_count = 3;
+            oaep = (CK_RSA_PKCS_OAEP_PARAMS *)mech->pParameter;
+            if (oaep == NULL ||
+                mech->ulParameterLen != sizeof(CK_RSA_PKCS_OAEP_PARAMS))
+                return CKR_MECHANISM_PARAM_INVALID;
+
+            if (oaep->source == CKZ_DATA_SPECIFIED &&
+                oaep->ulSourceDataLen > 0) {
+                TRACE_ERROR("CCA does not support non-empty OAEP source "
+                            "data\n");
```

Comment:
i would recommend to move the "data\n" to the previous line

Reply:
will fix